weblate/api/views.py

```diff
@@ -250,6 +252,52 @@ def repository(self, request, **kwargs):
         return Response(data)
 
 
+class UserViewSet(WeblateViewSet, CreateModelMixin, DestroyModelMixin):
+    """Users API."""
+
+    queryset = User.objects.none()
+    serializer_class = UserSerializer
+    lookup_field = "username"
+
+    def get_queryset(self):
+        return User.objects.all()
+
+    def partial_update(self, request, *args, **kwargs):
+        serializer_context = {
+            "request": request,
+        }
+        instance = User.objects.get(username=kwargs.get("username"))
+        serializer = self.serializer_class(
+            instance, data=request.data, partial=True, context=serializer_context
+        )
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data)
+
+
+class GroupViewSet(WeblateViewSet, CreateModelMixin, DestroyModelMixin):
+    """Groups API."""
+
+    queryset = Group.objects.none()
+    serializer_class = GroupSerializer
+    lookup_field = "name"
+
+    def get_queryset(self):
+        return Group.objects.all()
```

Comment:
Needs permission check, user should be probably able to only list own groups, all would be listed only with proper permission.

```suggestion
        if request.user.has_perm("group.edit"):
            return Group.objects.all()
        return request.user.groups.all()
```